packages/storage/compat/reference.ts

```diff
@@ -114,8 +115,17 @@ export class ReferenceCompat implements types.Reference {
     metadata?: Metadata
   ): types.UploadTask {
     this._throwIfRoot('putString');
+    const data = dataFromString(format, value);
```

Comment:
Just curious - could this method just call `put` with `new FbsBlob(data.data, true)`?

Reply:
You mean call `uploadBytes` instead of having a shared _nonResumableUpload function? I guess the only drawback would be an extra _throwIfRoot call that won't do anything.

Reply:
Yes